libraries/surplus/src/components.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import S from 's-js';
+import * as Surplus from 'surplus';
+import on from 'surplus-mixin-on';
+import 'ce-without-children';
+import 'ce-with-children';
+import 'ce-with-properties';
+import 'ce-with-event';
+
+export const ComponentWithoutChildren = () =>
+    <div>
+        <ce-without-children ref={__.wc}></ce-without-children>
+    </div>;
+
+export const ComponentWithChildren = () =>
+    <div>
+        <ce-with-children ref={__.wc}></ce-with-children>
+    </div>;
+
+export const ComponentWithChildrenRerender = () => {
+    const count = S.data(1);
+    Promise.resolve().then(() => count(count() + 1));
+    //S.cleanup(() =>  clearInterval(this.interval)); // what does this do? where does .interval come from?
```

Comment:
?

Reply:
? here as well :).  I started with the React impl, and it has this line: https://github.com/webcomponents/custom-elements-everywhere/blob/50fb30f0e8200ddb0743a3609a20636f998edfb3/libraries/react/src/components.js#L54 .  I couldn't see anywhere in the codebase that this.interval was being set?  Is that something the test harness uses?

Reply:
Wondering if that should be removed from the React version?